packages/flutter_tools/lib/src/commands/analyze_once.dart

```diff
@@ -194,4 +193,21 @@ class AnalyzeOnce extends AnalyzeBase {
       throwToolExit('Server error(s) occurred. (ran in ${seconds}s)');
     }
   }
+
+  bool _isFatal(List<AnalysisError> errors) {
+    for (final AnalysisError error in errors) {
+      final AnalysisSeverity severityLevel = error.writtenError.severityLevel;
+      if (severityLevel == AnalysisSeverity.error) {
+        return true;
+      }
+      if (severityLevel == AnalysisSeverity.warning &&
+        (argResults['fatal-warnings'] as bool || argResults['fatal-infos'] as bool)) {
```

Comment:
why does this need to check for both info and warning to be set? Seems like it should only be fatal-warnings

Reply:
warning severity level higher than info, see above experiments.
e.g. if fatal-infos, both error warning info will be fatal.

Reply:
Instead of having two flags, could we have one option that takes different severity levels? Like `--fatal=infos` ?

Reply:
Since error is always more fatal than info, if user has opted for fatal info, then it makes perfect sense to make it fatal for error too and also that feels like the expected behavior.